lib/oban/query.ex

```diff
@@ -105,6 +105,24 @@ defmodule Oban.Query do
     )
   end
 
+  @spec retry_job(Config.t(), pos_integer()) :: :ok
+  def retry_job(%Config{prefix: prefix, repo: repo, log: log}, id) do
+    Job
+    |> where([j], j.id == ^id)
+    |> where([j], j.state != "executing")
+    |> update([j],
+      set: [
+        state: "available",
+        max_attempts: fragment("GREATEST(?, ? + 1)", j.max_attempts, j.attempt),
+        completed_at: nil,
+        discarded_at: nil
+      ]
+    )
+    |> repo.update_all([], log: log, prefix: prefix)
+
```

Comment:
Questions: 
1. do I need to use any lock?
2. should I avoid `available` jobs to avoid the useless update?

Reply:
1. No need for a separate lock, the row-level lock is enough here.
2. It should avoid `scheduled`, `available` and `executing` jobs.

Reply:
There is a loose end for `retryable` jobs here as well. I think this should set `scheduled_at` to the current time as well.